RebbleOS/vibrate.c

```diff
@@ -23,18 +23,37 @@
 #include "vibrate.h"
 #include "task.h"
 #include "semphr.h"
+#include <stdbool.h>
 
+#define VIBRATE_QUEUE_MAX_WAIT_TICKS  20    // chosen by fair dice roll
 
-static uint16_t VIBRO_SHORT[] = { 255, 1000 };
-static uint16_t VIBRO_TAP_TAP[] = { 255, 750, 0, 750, 255, 750 };
+
+/*
+ * Initialization of default patterns.
+ * 
+ * TODO Maybe load these dinamically at some point
+ */
+static VibratePattern_t _defaultVibratePatterns[VIBRATE_CMD_MAX] = {
+    {.length =  2, .buffer = (const uint16_t []) {255, 1000}},                   // VIBRATE_CMD_PLAY_PATTERN_1
+    {.length =  6, .buffer = (const uint16_t []) {255, 750, 0, 750, 255, 750}},  // VIBRATE_CMD_PALY_PATTERN_2
+    {.length = 30, .buffer = (const uint16_t []) {255, 750, 0, 750, 255, 750,
+                                                  255, 750, 0, 750, 255, 750, 
+                                                  255, 750, 0, 750, 255, 750, 
+                                                  255, 750, 0, 750, 255, 750, 
+                                                  255, 750, 0, 750, 255, 750}},  // VIBRATE_CMD_PALY_PATTERN_3
+    {.length =  0, .buffer = (const uint16_t []) {}},                          // VIBRATE_CMD_STOP
+};
 
 static TaskHandle_t _vibrate_task;
 static xQueueHandle _vibrate_queue;
+static VibratePattern_t *currentPattern;
```

Comment:
only one `current_pattern`, but it was initialized in `_vibrate_thread`. Fixed now (see 2 items below)